prow/cluster/monitoring/prow_alertmanager.yaml

```diff
@@ -14,3 +14,14 @@ spec:
     runAsUser: 1000
   serviceAccountName: alertmanager
   version: v0.15.2
+  storage: # Note that this section is immutable so changes require deleting and recreating the resource.
```

Comment:
There's no better alternative? This sounds like a maintenance hassle.

Reply:
Yeah it is. We had to do something similar when changing some Prow services from `NodePort` to `ClusterIP` since that is also an immutable field.

Can the bazel rules be configured to do a replace instead of apply for just one component? If so that could work, but I'm also not sure that we want to use replace on this resource (or all the other partially immutable resource this applies to) unless we need to change an immutable section. The delete and create operations might cause some downtime while the prometheus operator reacts whereas an apply wouldn't. So conditionally replacing instead of applying or if applying fails would be ideal.

FWIW I don't think this particular section will need to be changed once we have it set up.

Reply:
I don't think there's that level of granularity